php/tests/wrapper_type_setters_test.php

```diff
@@ -0,0 +1,221 @@
+    {
+        return [
+            [TestWrapperSetters::class, "setDoubleValueValue", "abc"],
+            [TestWrapperSetters::class, "setDoubleValueValue", []],
+            [TestWrapperSetters::class, "setDoubleValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setDoubleValueValue", new DoubleValue()],
+
+            [TestWrapperSetters::class, "setFloatValueValue", "abc"],
+            [TestWrapperSetters::class, "setFloatValueValue", []],
+            [TestWrapperSetters::class, "setFloatValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setFloatValueValue", new FloatValue()],
+
+            [TestWrapperSetters::class, "setInt64ValueValue", "abc"],
+            [TestWrapperSetters::class, "setInt64ValueValue", []],
+            [TestWrapperSetters::class, "setInt64ValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setInt64ValueValue", new Int64Value()],
+
+            [TestWrapperSetters::class, "setUInt64ValueValue", "abc"],
+            [TestWrapperSetters::class, "setUInt64ValueValue", []],
+            [TestWrapperSetters::class, "setUInt64ValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setUInt64ValueValue", new UInt64Value()],
+
+            [TestWrapperSetters::class, "setInt32ValueValue", "abc"],
+            [TestWrapperSetters::class, "setInt32ValueValue", []],
+            [TestWrapperSetters::class, "setInt32ValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setInt32ValueValue", new Int32Value()],
+
+            [TestWrapperSetters::class, "setUInt32ValueValue", "abc"],
+            [TestWrapperSetters::class, "setUInt32ValueValue", []],
+            [TestWrapperSetters::class, "setUInt32ValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setUInt32ValueValue", new UInt32Value()],
+
+            [TestWrapperSetters::class, "setBoolValueValue", []],
+            [TestWrapperSetters::class, "setBoolValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setBoolValueValue", new BoolValue()],
+
+            [TestWrapperSetters::class, "setStringValueValue", []],
+            [TestWrapperSetters::class, "setStringValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setStringValueValue", new StringValue()],
+
+            [TestWrapperSetters::class, "setBytesValueValue", []],
+            [TestWrapperSetters::class, "setBytesValueValue", new stdClass()],
+            [TestWrapperSetters::class, "setBytesValueValue", new BytesValue()],
+        ];
+    }
+
+    /**
+     * @dataProvider constructorWithWrapperTypeDataProvider
+     */
+    public function testConstructorWithWrapperType($class, $wrapperClass, $wrapperField, $value)
+    {
+        $actualInstance = new $class([$wrapperField => $value]);
+        $expectedInstance = new $class([$wrapperField => new $wrapperClass(['value' => $value])]);
+        $this->assertEquals($expectedInstance, $actualInstance);
```

Comment:
Currently, assertEquals doesn't work for message in c extension. Basically, the assertEquals assert nothing in c extension. Could you actually test equal of value?